rq_scheduler/scheduler.py

```diff
@@ -322,6 +322,35 @@ def count(self, until=None):
         until = rationalize_until(until)
         return self.connection.zcount(self.scheduled_jobs_key, 0, until)
 
+    def _get_job_from_id(self, job_id):
+        """
+        Fetch the job with given job ID or return None if no matching job
+        exists.
+        """
+        try:
+            return self.job_class.fetch(job_id, connection=self.connection)
+        except NoSuchJobError:
+            # Delete jobs that aren't there from scheduler
+            self.cancel(job_id)
```

Comment:
[Explanatory] This function contains code factored out of `get_jobs` because both that function and `get_job` need this functionality.